src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -13,318 +13,332 @@ import IndexPatternsFieldListProvider from 'ui/index_patterns/_field_list';
 import IndexPatternsFlattenHitProvider from 'ui/index_patterns/_flatten_hit';
 import IndexPatternsCalculateIndicesProvider from 'ui/index_patterns/_calculate_indices';
 import IndexPatternsPatternCacheProvider from 'ui/index_patterns/_pattern_cache';
-export default function IndexPatternFactory(Private, timefilter, Notifier, config, kbnIndex, Promise, safeConfirm) {
-  let fieldformats = Private(RegistryFieldFormatsProvider);
-  let getIds = Private(IndexPatternsGetIdsProvider);
-  let mapper = Private(IndexPatternsMapperProvider);
-  let intervals = Private(IndexPatternsIntervalsProvider);
-  let DocSource = Private(DocSourceProvider);
-  let mappingSetup = Private(UtilsMappingSetupProvider);
-  let FieldList = Private(IndexPatternsFieldListProvider);
 
-  let flattenHit = Private(IndexPatternsFlattenHitProvider);
-  let calculateIndices = Private(IndexPatternsCalculateIndicesProvider);
-  let patternCache = Private(IndexPatternsPatternCacheProvider);
-
-  let type = 'index-pattern';
-
-  let notify = new Notifier();
-
-  let mapping = mappingSetup.expandShorthand({
+export default function IndexPatternFactory(Private, Notifier, config, kbnIndex, Promise, safeConfirm) {
+  const fieldformats = Private(RegistryFieldFormatsProvider);
+  const getIds = Private(IndexPatternsGetIdsProvider);
+  const mapper = Private(IndexPatternsMapperProvider);
+  const intervals = Private(IndexPatternsIntervalsProvider);
+  const DocSource = Private(DocSourceProvider);
+  const mappingSetup = Private(UtilsMappingSetupProvider);
+  const FieldList = Private(IndexPatternsFieldListProvider);
+  const flattenHit = Private(IndexPatternsFlattenHitProvider);
+  const calculateIndices = Private(IndexPatternsCalculateIndicesProvider);
+  const patternCache = Private(IndexPatternsPatternCacheProvider);
+  const type = 'index-pattern';
+  const notify = new Notifier();
+
+  const mapping = mappingSetup.expandShorthand({
     title: 'string',
     timeFieldName: 'string',
     notExpandable: 'boolean',
     intervalName: 'string',
     fields: 'json',
     fieldFormatMap: {
       type: 'string',
-      _serialize: function (map) {
-        if (map == null) return;
-
-        let count = 0;
-        let serialized = _.transform(map, function (flat, format, field) {
-          if (!format) return;
-          count++;
-          flat[field] = format;
-        });
-
-        if (count) return angular.toJson(serialized);
+      _serialize(map = {}) {
+        const serialized = _.transform(map, serialize);
```

Comment:
I'm in favor of simple code over "excruciatingly correct" code, because maintainability. That said I can change it if you guys are adamant about it. Maybe @epixa can weigh in as well.